mlbench_core/optim/pytorch/optim.py

```diff
@@ -499,12 +499,290 @@ def step(self, closure=None):
         return loss
 
 
+class PowerSGD(Optimizer):
```

Comment:
Great work. One small comment: Is it possible to separate the aggregation scheme from the Optimizer itself ? Like instead of defining an `all_reduce`, maybe create a new aggregator ? Would be useful in case we want to re-use it for different optimizers.

Reply:
I think that those `all_reduce` and `all_gather` functions in the global scope can be removed, but I don't see why you would want to use other aggregators than `torch.distributed.all_reduce`. Do you already have other instances of that in the codebase?

Reply:
@ehoelzl Did you mean moving the `reduce` function from PowerSGD to a new aggregator?